libdevcore/Assertions.h

```diff
@@ -73,7 +73,7 @@ inline bool assertEqualAux(A const& _a, B const& _b, char const* _aStr, char con
 /// Use it as assertThrow(1 == 1, ExceptionType, "Mathematics is wrong.");
 /// Do NOT supply an exception object as the second parameter.
 #define assertThrow(_condition, _ExceptionType, _description) \
-	::dev::assertThrowAux<_ExceptionType>(_condition, _description, __LINE__, __FILE__, ETH_FUNC)
+	::dev::assertThrowAux<_ExceptionType>(bool(_condition), _description, __LINE__, __FILE__, ETH_FUNC)
```

Comment:
We usually use `assert(!!x, "msg")` if the compiler complains. Is there a situation where `bool(...)` is more permissive than we would like?